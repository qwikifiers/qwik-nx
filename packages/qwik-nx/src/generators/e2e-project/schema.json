{
  "$schema": "http://json-schema.org/schema",
  "$id": "E2eProject",
  "title": "Create an E2E app for a Qwik app",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "The name of the frontend project to test."
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["playwright", "cypress"],
      "description": "Test runner to use for end to end (E2E) tests.",
      "default": "playwright"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    }
  },
  "required": ["project", "e2eTestRunner"]
}
