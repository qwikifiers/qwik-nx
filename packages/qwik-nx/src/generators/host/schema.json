{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Host",
  "title": "Generate Micro Frontends Setup for Qwik Host App",
  "description": "Create Micro Frontends configuration files for the Qwik Application.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "x-priority": "important"
    },
    "style": {
      "description": "The file extension to be used for style files.",
      "type": "string",
      "default": "css",
      "alias": "s",
      "x-prompt": {
        "message": "Which stylesheet format would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "css",
            "label": "CSS"
          },
          {
            "value": "scss",
            "label": "SASS(.scss)       [ http://sass-lang.com                     ]"
          },
          {
            "value": "styl",
            "label": "Stylus(.styl)     [ http://stylus-lang.com                   ]"
          },
          {
            "value": "less",
            "label": "LESS              [ http://lesscss.org                       ]"
          }
        ],
        "default": "css"
      }
    },
    "linter": {
      "description": "The tool to use for running lint checks.",
      "type": "string",
      "enum": ["eslint", "none"],
      "default": "eslint"
    },
    "skipFormat": {
      "description": "Skip formatting files.",
      "type": "boolean",
      "default": false
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["vitest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "vitest"
    },
    "strict": {
      "type": "boolean",
      "description": "Creates an application with strict mode and strict type checking.",
      "default": true
    },
    "tailwind": {
      "description": "Setup Tailwind",
      "type": "boolean",
      "default": false
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["playwright", "cypress", "none"],
      "description": "Test runner to use for end to end (E2E) tests.",
      "default": "playwright"
    },
    "port": {
      "type": "number",
      "description": "The port for the dev and preview servers of the app (they're the same by default in MF setup).",
      "default": 4200
    },
    "remotes": {
      "type": "array",
      "description": "A list of remote application names that the host application should consume.",
      "x-priority": "important"
    }
  },
  "required": ["name"]
}
