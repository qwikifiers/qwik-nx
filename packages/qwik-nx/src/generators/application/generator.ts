import {
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  names,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import { NormalizedSchema, normalizeOptions } from './utils/normalize-options';
import { viteConfigurationGenerator } from '@nrwl/vite';
import { QwikAppGeneratorSchema } from './schema';
import { Linter } from '@nrwl/linter';
import { addStyledModuleDependencies } from '../../utils/add-styled-dependencies';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import { configureEslint } from '../../utils/configure-eslint';
import { addCommonQwikDependencies } from '../../utils/add-common-qwik-dependencies';

function addFiles(tree: Tree, options: NormalizedSchema) {
  const templateOptions = {
    ...options,
    ...names(options.name),
    template: '',
  };
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    options.appProjectRoot,
    templateOptions
  );
}

async function configureVite(
  tree: Tree,
  options: NormalizedSchema
): Promise<GeneratorCallback> {
  const tsConfigPath = joinPathFragments(
    options.appProjectRoot,
    'tsconfig.json'
  );
  const tsConfig = tree.read(tsConfigPath);
  const includeVitest = options.unitTestRunner === 'vitest';

  const callback = await viteConfigurationGenerator(tree, {
    project: options.projectName,
    newProject: false,
    includeVitest,
    includeLib: false,
    uiFramework: 'none',
    inSourceTests: false,
  });
  tree.delete(joinPathFragments(options.appProjectRoot, 'index.html'));

  // overwrite tsconfig back after "viteConfigurationGenerator"
  tree.write(tsConfigPath, tsConfig);

  if (includeVitest) {
    const projectConfig = readProjectConfiguration(tree, options.projectName);
    const testTarget = projectConfig.targets['test'];
    testTarget.outputs = ['{workspaceRoot}/coverage/{projectRoot}'];
    updateProjectConfiguration(tree, options.projectName, projectConfig);
  }

  return callback;
}

export default async function (tree: Tree, options: QwikAppGeneratorSchema) {
  const normalizedOptions = normalizeOptions(tree, options);
  const tasks: GeneratorCallback[] = [];

  addProjectConfiguration(tree, normalizedOptions.projectName, {
    root: normalizedOptions.appProjectRoot,
    name: normalizedOptions.projectName,
    projectType: 'application',
    sourceRoot: `${normalizedOptions.appProjectRoot}/src`,
    // TODO: revisit later, now "viteConfigurationGenerator" will fail if there's no build target specified
    targets: { build: { options: {} } },
    tags: normalizedOptions.parsedTags,
  });

  addFiles(tree, normalizedOptions);

  tasks.push(await configureVite(tree, normalizedOptions));

  if (normalizedOptions.linter === Linter.EsLint) {
    tasks.push(configureEslint(tree, normalizedOptions.projectName, true));
  }

  if (normalizedOptions.style !== 'none') {
    tasks.push(
      addStyledModuleDependencies(tree, normalizedOptions.styleExtension)
    );
  }

  tasks.push(addCommonQwikDependencies(tree));

  if (!options.skipFormat) {
    await formatFiles(tree);
  }

  return runTasksInSerial(...tasks);
}
