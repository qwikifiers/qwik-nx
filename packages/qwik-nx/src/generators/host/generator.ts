import {
  formatFiles,
  generateFiles,
  GeneratorCallback,
  joinPathFragments,
  readProjectConfiguration,
  Tree,
  updateProjectConfiguration,
} from '@nrwl/devkit';
import { HostGeneratorSchema } from './schema';
import { appGenerator } from './../application/generator';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import remoteGenerator from '../remote/generator';
import { normalizeOptions } from '../application/utils/normalize-options';
import { join } from 'path';
import { addMicroFrontendBetaWarning } from '../../utils/mf-beta-warning';
import { NormalizedSchema } from '../application/schema';

function addFiles(
  tree: Tree,
  schema: NormalizedSchema,
  remotes: { name: string; port: number }[]
) {
  // TODO: switch to empty preset once available
  tree.delete(
    joinPathFragments(schema.projectRoot, 'src/routes/flower/index.tsx')
  );
  tree.delete(
    joinPathFragments(schema.projectRoot, 'src/routes/flower/flower.css')
  );

  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files'),
    schema.projectRoot,
    {
      name: schema.projectName,
      remotes: remotes.map((r) => r.name),
    }
  );

  tree.write(
    join(schema.projectRoot, 'src/config/remotes.json'),
    JSON.stringify(
      Object.fromEntries(
        remotes.map((r) => [r.name, `http://localhost:${r.port}`])
      )
    )
  );

  const projectConfig = readProjectConfiguration(tree, schema.projectName);
  projectConfig.targets!['serve'].executor =
    'qwik-nx:micro-frontends-dev-server';
  projectConfig.targets!['preview'].executor =
    'qwik-nx:micro-frontends-preview-server';
  updateProjectConfiguration(tree, schema.projectName, projectConfig);
}

export async function hostGenerator(tree: Tree, options: HostGeneratorSchema) {
  addMicroFrontendBetaWarning();
  const tasks: GeneratorCallback[] = [];

  const normalizedSchema = normalizeOptions(tree, {
    ...options,
    devServerPort: options.port,
    previewServerPort: options.port,
  });

  const initTask = await appGenerator(tree, {
    ...normalizedSchema,
    skipFormat: true,
  });
  tasks.push(initTask);

  const remotesWithPorts: {
    name: string;
    port: number;
  }[] = [];

  if (options.remotes) {
    let port = normalizedSchema.devServerPort + 1;
    for (const remote of options.remotes) {
      remotesWithPorts.push({
        name: remote,
        port,
      });

      await remoteGenerator(tree, {
        ...options,
        name: remote,
        port,
        skipFormat: true,
      });
      port++;
    }
  }

  addFiles(tree, normalizedSchema, remotesWithPorts);

  if (!options.skipFormat) {
    await formatFiles(tree);
  }
  return runTasksInSerial(...tasks);
}

export default hostGenerator;
