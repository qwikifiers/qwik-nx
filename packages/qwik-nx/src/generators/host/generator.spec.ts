import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree, readProjectConfiguration, readJson } from '@nx/devkit';

import generator from './generator';
import { HostGeneratorSchema } from './schema';
import { getFormattedListChanges } from '../../utils/testing-generators';

describe('host generator', () => {
  let appTree: Tree;
  const options: HostGeneratorSchema = {
    name: 'myhostapp',
    remotes: ['remote1', 'remote2'],
  };

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  });

  it('should run successfully', async () => {
    await generator(appTree, options);
    const config = readProjectConfiguration(appTree, 'myhostapp');
    expect(config).toBeDefined();

    expect(readJson(appTree, 'apps/myhostapp/src/config/remotes.json')).toEqual(
      {
        remote1: 'http://localhost:4201',
        remote2: 'http://localhost:4202',
      }
    );

    // host snapshots
    expect(
      appTree.read('apps/myhostapp/vite.config.ts', 'utf-8')
    ).toMatchSnapshot();
    expect(
      appTree.read('apps/myhostapp/src/entry.ssr.tsx', 'utf-8')
    ).toMatchSnapshot();
    expect(
      appTree.read('apps/myhostapp/project.json', 'utf-8')
    ).toMatchSnapshot();
    expect(getFormattedListChanges(appTree)).toMatchSnapshot();

    // remote snapshots
    for (const remote of ['remote1', 'remote2']) {
      expect(
        appTree.read(`apps/${remote}/vite.config.ts`, 'utf-8')
      ).toMatchSnapshot();
      expect(
        appTree.read(`apps/${remote}/src/entry.ssr.tsx`, 'utf-8')
      ).toMatchSnapshot();
      expect(
        appTree.read(`apps/${remote}/project.json`, 'utf-8')
      ).toMatchSnapshot();
    }
  });
});
