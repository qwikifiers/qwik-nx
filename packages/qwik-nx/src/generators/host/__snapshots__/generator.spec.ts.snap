// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`host generator should run successfully 1`] = `
"import { qwikVite } from '@builder.io/qwik/optimizer';
import { qwikCity } from '@builder.io/qwik-city/vite';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';
import { qwikNxVite } from 'qwik-nx/plugins';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/apps/myhostapp',
  plugins: [
    qwikNxVite(),
    qwikCity(), 
    qwikVite({
			client: {
				outDir: '../../dist/apps/myhostapp/client',
			},
			ssr: {
				outDir: '../../dist/apps/myhostapp/server',
			},
		}),
    tsconfigPaths({ root: '../../' })
  ],
  server: {
    fs: {
      // Allow serving files from the project root
      allow: ['../../'],
    },
  },
  preview: {
    headers: {
      'Cache-Control': 'public, max-age=600',
    },
  },
   test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'node',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}']
  }
  
});
"
`;

exports[`host generator should run successfully 2`] = `
"/**
 * WHAT IS THIS FILE?
 *
 * SSR entry point, in all cases the application is render outside the browser, this
 * entry point will be the common one.
 *
 * - Server (express, cloudflare...)
 * - npm run start
 * - npm run preview
 * - npm run build
 *
 */
import { renderToStream, RenderToStreamOptions } from '@builder.io/qwik/server';
import { manifest } from '@qwik-client-manifest';
import Root from './root';

export default function (opts: RenderToStreamOptions) {
  return renderToStream(<Root />, {
    manifest,
    ...opts,
    // Use container attributes to set attributes on the html tag.
    containerAttributes: {
      lang: 'en-us',
      ...opts.containerAttributes,
    },
    prefetchStrategy: {
      implementation: {
        linkInsert: null,
        workerFetchInsert: null,
        prefetchEvent: 'always',
      },
    },
    qwikLoader: {
      include: 'always',
    },
    streaming: {
      inOrder: {
        strategy: 'auto',
        maximunInitialChunk: 0,
        maximunChunk: 0,
      },
    },
  });
}
"
`;

exports[`host generator should run successfully 3`] = `
"{
  \\"name\\": \\"myhostapp\\",
  \\"$schema\\": \\"../../node_modules/nx/schemas/project-schema.json\\",
  \\"projectType\\": \\"application\\",
  \\"sourceRoot\\": \\"apps/myhostapp/src\\",
  \\"targets\\": {
    \\"build\\": {
      \\"executor\\": \\"qwik-nx:build\\",
      \\"options\\": {
        \\"runSequence\\": [
          \\"myhostapp:build.client\\",
          \\"myhostapp:build.ssr\\"
        ],
        \\"outputPath\\": \\"dist/apps/myhostapp\\"
      },
      \\"configurations\\": {
        \\"preview\\": {}
      }
    },
    \\"build.client\\": {
      \\"executor\\": \\"@nrwl/vite:build\\",
      \\"options\\": {
        \\"outputPath\\": \\"dist/apps/myhostapp\\",
        \\"configFile\\": \\"apps/myhostapp/vite.config.ts\\"
      }
    },
    \\"build.ssr\\": {
      \\"executor\\": \\"@nrwl/vite:build\\",
      \\"defaultConfiguration\\": \\"preview\\",
      \\"options\\": {
        \\"outputPath\\": \\"dist/apps/myhostapp\\"
      },
      \\"configurations\\": {
        \\"preview\\": {
          \\"ssr\\": \\"src/entry.preview.tsx\\",
          \\"mode\\": \\"production\\"
        }
      }
    },
    \\"preview\\": {
      \\"executor\\": \\"qwik-nx:micro-frontends-preview-server\\",
      \\"options\\": {
        \\"buildTarget\\": \\"myhostapp:build\\",
        \\"port\\": 4300
      }
    },
    \\"test\\": {
      \\"executor\\": \\"@nrwl/vite:test\\",
      \\"outputs\\": [
        \\"../../coverage/apps/myhostapp\\"
      ],
      \\"options\\": {
        \\"passWithNoTests\\": true,
        \\"reportsDirectory\\": \\"../../coverage/apps/myhostapp\\"
      }
    },
    \\"serve\\": {
      \\"executor\\": \\"qwik-nx:micro-frontends-dev-server\\",
      \\"options\\": {
        \\"buildTarget\\": \\"myhostapp:build.client\\",
        \\"mode\\": \\"ssr\\",
        \\"port\\": 4200
      }
    },
    \\"serve.debug\\": {
      \\"executor\\": \\"nx:run-commands\\",
      \\"options\\": {
        \\"command\\": \\"node --inspect-brk ../../node_modules/vite/bin/vite.js --mode ssr --force\\",
        \\"cwd\\": \\"apps/myhostapp\\"
      }
    }
  },
  \\"tags\\": []
}
"
`;

exports[`host generator should run successfully 4`] = `
Array [
  Object {
    "path": ".prettierrc",
    "type": "CREATE",
  },
  Object {
    "path": "package.json",
    "type": "CREATE",
  },
  Object {
    "path": "nx.json",
    "type": "CREATE",
  },
  Object {
    "path": "tsconfig.base.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "libs/.gitignore",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/.eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/.prettierignore",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/README.md",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/package.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/public/favicon.svg",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/public/manifest.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/public/robots.txt",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/components/header/header.,",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/components/header/header.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/components/icons/qwik.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/components/router-head/router-head.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/entry.dev.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/entry.preview.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/entry.ssr.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/global.,",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/root.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/routes/flower/flower.,",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/routes/index.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/routes/layout.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/routes/service-worker.ts",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/tsconfig.app.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/tsconfig.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/tsconfig.spec.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/vite.config.ts",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/.eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/.prettierignore",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/README.md",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/package.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/public/favicon.svg",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/public/manifest.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/public/robots.txt",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/src/components/header/header.,",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/src/entry.dev.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/src/entry.preview.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/src/entry.ssr.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/src/global.,",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/src/root.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/src/routes/index.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/src/routes/layout.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/src/routes/service-worker.ts",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/tsconfig.app.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/tsconfig.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/tsconfig.spec.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/vite.config.ts",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/src/components/.gitkeep",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote1/src/routes/about/index.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/project.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/.eslintrc.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/.prettierignore",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/README.md",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/package.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/public/favicon.svg",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/public/manifest.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/public/robots.txt",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/src/components/header/header.,",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/src/entry.dev.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/src/entry.preview.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/src/entry.ssr.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/src/global.,",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/src/root.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/src/routes/index.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/src/routes/layout.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/src/routes/service-worker.ts",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/tsconfig.app.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/tsconfig.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/tsconfig.spec.json",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/vite.config.ts",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/src/components/.gitkeep",
    "type": "CREATE",
  },
  Object {
    "path": "apps/remote2/src/routes/about/index.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/components/remote-mfe/remote-mfe.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/routes/about/index.tsx",
    "type": "CREATE",
  },
  Object {
    "path": "apps/myhostapp/src/config/remotes.json",
    "type": "CREATE",
  },
]
`;

exports[`host generator should run successfully 5`] = `
"import { qwikVite } from '@builder.io/qwik/optimizer';
import { qwikCity } from '@builder.io/qwik-city/vite';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';
import { qwikNxVite } from 'qwik-nx/plugins';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/apps/remote1',
  plugins: [
    qwikNxVite(),
    qwikCity(), 
    qwikVite({
			client: {
				outDir: '../../dist/apps/remote1/client',
			},
			ssr: {
				outDir: '../../dist/apps/remote1/server',
			},
		}),
    tsconfigPaths({ root: '../../' })
  ],
  server: {
    fs: {
      // Allow serving files from the project root
      allow: ['../../'],
    },
  },
  preview: {
    headers: {
      'Cache-Control': 'public, max-age=600',
    },
  },
   test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'node',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}']
  }
  
});
"
`;

exports[`host generator should run successfully 6`] = `
"/**
 * WHAT IS THIS FILE?
 *
 * SSR entry point, in all cases the application is render outside the browser, this
 * entry point will be the common one.
 *
 * - Server (express, cloudflare...)
 * - npm run start
 * - npm run preview
 * - npm run build
 *
 */
import { renderToStream, RenderToStreamOptions } from '@builder.io/qwik/server';
import { manifest } from '@qwik-client-manifest';
import Root from './root';

export default function (opts: RenderToStreamOptions) {
  return renderToStream(<Root />, {
    manifest,
    ...opts,
    // Use container attributes to set attributes on the html tag.
    containerAttributes: {
      lang: 'en-us',
      ...opts.containerAttributes,
    },
    base: (opts) => {
      // removing query params from url and unifying to the form \\"http://localhost:5001/build/\\"
      return new URL(opts.serverData!.url).origin + '/build/';
    },
    prefetchStrategy: {
      implementation: {
        linkInsert: null,
        workerFetchInsert: null,
        prefetchEvent: 'always',
      },
    },
    containerTagName: 'div',
    qwikLoader: {
      include: 'never',
    },
  });
}
"
`;

exports[`host generator should run successfully 7`] = `
"{
  \\"name\\": \\"remote1\\",
  \\"$schema\\": \\"../../node_modules/nx/schemas/project-schema.json\\",
  \\"projectType\\": \\"application\\",
  \\"sourceRoot\\": \\"apps/remote1/src\\",
  \\"targets\\": {
    \\"build\\": {
      \\"executor\\": \\"qwik-nx:build\\",
      \\"options\\": {
        \\"runSequence\\": [
          \\"remote1:build.client\\",
          \\"remote1:build.ssr\\"
        ],
        \\"outputPath\\": \\"dist/apps/remote1\\"
      },
      \\"configurations\\": {
        \\"preview\\": {}
      }
    },
    \\"build.client\\": {
      \\"executor\\": \\"@nrwl/vite:build\\",
      \\"options\\": {
        \\"outputPath\\": \\"dist/apps/remote1\\",
        \\"configFile\\": \\"apps/remote1/vite.config.ts\\"
      }
    },
    \\"build.ssr\\": {
      \\"executor\\": \\"@nrwl/vite:build\\",
      \\"defaultConfiguration\\": \\"preview\\",
      \\"options\\": {
        \\"outputPath\\": \\"dist/apps/remote1\\"
      },
      \\"configurations\\": {
        \\"preview\\": {
          \\"ssr\\": \\"src/entry.preview.tsx\\",
          \\"mode\\": \\"production\\"
        }
      }
    },
    \\"preview\\": {
      \\"executor\\": \\"@nrwl/vite:preview-server\\",
      \\"options\\": {
        \\"buildTarget\\": \\"remote1:build\\",
        \\"port\\": 4201
      }
    },
    \\"test\\": {
      \\"executor\\": \\"@nrwl/vite:test\\",
      \\"outputs\\": [
        \\"../../coverage/apps/remote1\\"
      ],
      \\"options\\": {
        \\"passWithNoTests\\": true,
        \\"reportsDirectory\\": \\"../../coverage/apps/remote1\\"
      }
    },
    \\"serve\\": {
      \\"executor\\": \\"@nrwl/vite:dev-server\\",
      \\"options\\": {
        \\"buildTarget\\": \\"remote1:build.client\\",
        \\"mode\\": \\"ssr\\",
        \\"port\\": 4201
      }
    },
    \\"serve.debug\\": {
      \\"executor\\": \\"nx:run-commands\\",
      \\"options\\": {
        \\"command\\": \\"node --inspect-brk ../../node_modules/vite/bin/vite.js --mode ssr --force\\",
        \\"cwd\\": \\"apps/remote1\\"
      }
    }
  },
  \\"tags\\": []
}
"
`;

exports[`host generator should run successfully 8`] = `
"import { qwikVite } from '@builder.io/qwik/optimizer';
import { qwikCity } from '@builder.io/qwik-city/vite';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';
import { qwikNxVite } from 'qwik-nx/plugins';

export default defineConfig({
  cacheDir: '../../node_modules/.vite/apps/remote2',
  plugins: [
    qwikNxVite(),
    qwikCity(), 
    qwikVite({
			client: {
				outDir: '../../dist/apps/remote2/client',
			},
			ssr: {
				outDir: '../../dist/apps/remote2/server',
			},
		}),
    tsconfigPaths({ root: '../../' })
  ],
  server: {
    fs: {
      // Allow serving files from the project root
      allow: ['../../'],
    },
  },
  preview: {
    headers: {
      'Cache-Control': 'public, max-age=600',
    },
  },
   test: {
    globals: true,
    cache: {
      dir: '../../node_modules/.vitest',
    },
    environment: 'node',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}']
  }
  
});
"
`;

exports[`host generator should run successfully 9`] = `
"/**
 * WHAT IS THIS FILE?
 *
 * SSR entry point, in all cases the application is render outside the browser, this
 * entry point will be the common one.
 *
 * - Server (express, cloudflare...)
 * - npm run start
 * - npm run preview
 * - npm run build
 *
 */
import { renderToStream, RenderToStreamOptions } from '@builder.io/qwik/server';
import { manifest } from '@qwik-client-manifest';
import Root from './root';

export default function (opts: RenderToStreamOptions) {
  return renderToStream(<Root />, {
    manifest,
    ...opts,
    // Use container attributes to set attributes on the html tag.
    containerAttributes: {
      lang: 'en-us',
      ...opts.containerAttributes,
    },
    base: (opts) => {
      // removing query params from url and unifying to the form \\"http://localhost:5001/build/\\"
      return new URL(opts.serverData!.url).origin + '/build/';
    },
    prefetchStrategy: {
      implementation: {
        linkInsert: null,
        workerFetchInsert: null,
        prefetchEvent: 'always',
      },
    },
    containerTagName: 'div',
    qwikLoader: {
      include: 'never',
    },
  });
}
"
`;

exports[`host generator should run successfully 10`] = `
"{
  \\"name\\": \\"remote2\\",
  \\"$schema\\": \\"../../node_modules/nx/schemas/project-schema.json\\",
  \\"projectType\\": \\"application\\",
  \\"sourceRoot\\": \\"apps/remote2/src\\",
  \\"targets\\": {
    \\"build\\": {
      \\"executor\\": \\"qwik-nx:build\\",
      \\"options\\": {
        \\"runSequence\\": [
          \\"remote2:build.client\\",
          \\"remote2:build.ssr\\"
        ],
        \\"outputPath\\": \\"dist/apps/remote2\\"
      },
      \\"configurations\\": {
        \\"preview\\": {}
      }
    },
    \\"build.client\\": {
      \\"executor\\": \\"@nrwl/vite:build\\",
      \\"options\\": {
        \\"outputPath\\": \\"dist/apps/remote2\\",
        \\"configFile\\": \\"apps/remote2/vite.config.ts\\"
      }
    },
    \\"build.ssr\\": {
      \\"executor\\": \\"@nrwl/vite:build\\",
      \\"defaultConfiguration\\": \\"preview\\",
      \\"options\\": {
        \\"outputPath\\": \\"dist/apps/remote2\\"
      },
      \\"configurations\\": {
        \\"preview\\": {
          \\"ssr\\": \\"src/entry.preview.tsx\\",
          \\"mode\\": \\"production\\"
        }
      }
    },
    \\"preview\\": {
      \\"executor\\": \\"@nrwl/vite:preview-server\\",
      \\"options\\": {
        \\"buildTarget\\": \\"remote2:build\\",
        \\"port\\": 4202
      }
    },
    \\"test\\": {
      \\"executor\\": \\"@nrwl/vite:test\\",
      \\"outputs\\": [
        \\"../../coverage/apps/remote2\\"
      ],
      \\"options\\": {
        \\"passWithNoTests\\": true,
        \\"reportsDirectory\\": \\"../../coverage/apps/remote2\\"
      }
    },
    \\"serve\\": {
      \\"executor\\": \\"@nrwl/vite:dev-server\\",
      \\"options\\": {
        \\"buildTarget\\": \\"remote2:build.client\\",
        \\"mode\\": \\"ssr\\",
        \\"port\\": 4202
      }
    },
    \\"serve.debug\\": {
      \\"executor\\": \\"nx:run-commands\\",
      \\"options\\": {
        \\"command\\": \\"node --inspect-brk ../../node_modules/vite/bin/vite.js --mode ssr --force\\",
        \\"cwd\\": \\"apps/remote2\\"
      }
    }
  },
  \\"tags\\": []
}
"
`;
